src/shared/services/auth/auth.js

```diff
@@ -244,9 +310,14 @@ export function logout() {
  *
  */
 export async function initAuth() {
-  restoreAccessToken(); // Restore acces token from local storage
+  restoreAccessToken(); // Restore access token from local storage
   catchError(); // Catch any error from the OAuth2 authorization service
   await handleCallback(); // Handle a callback from the OAuth2 authorization service
+
+  if (getOauthDomain() === 'keycloak') {
+    await handleRefreshAccessToken(); // Refresh access token if expired
+    setRefreshAccessTokenInterval(); // Start periodic check for validity of access token, and refresh if expired
```

Comment:
Ideally, we would refresh access tokens silently on demand (e.g. refresh tokens when a user requests data but the access token is expired). 

However, this is impractical in this application because (1) UI authentication state is directly dependent on the access_token expiry time and (2) we use rotating refresh tokens (i.e. with a 30 min lifespan) so if we don't refresh often, the user will be logged out after 30 minutes of not using the app.